picoCTF-shell/hacksport/deploy.py

```diff
@@ -65,19 +75,21 @@ def give_port():
         return inv_port_map[(context["problem"], context["instance"])]
 
     if len(context["port_map"].items()) + len(
-            context["config"].banned_ports_parsed) == 65536:
+            context["config"].banned_ports_parsed) == HIGHEST_PORT + 1:
         raise Exception(
             "All usable ports are taken. Cannot deploy any more instances.")
 
-    while True:
-        port = port_random.randint(0, 65535)
-        if port not in context["config"].banned_ports_parsed:
-            owner, instance = context["port_map"].get(port, (None, None))
-            if owner is None or (owner == context["problem"] and
-                                 instance == context["instance"]):
-                context["port_map"][port] = (context["problem"],
-                                             context["instance"])
-                return port
+    # collect list of banned ports
+    banned_ports_list = []
```

Comment:
So at this point `context["config"].banned_ports_parsed` or `banned_ports_result ` aren't really in play, but on other hand it's kinda redundant now with both the LOW and HIGH range declared and any actual exceptions such as `4242` would be caught in the `check_if_port_in_use` check.

Reply:
Not necessarily. context["config"].banned_ports_parsed contains the ports that were specified by the config file. It is possible that that file has ports in it that are not NOW or HIGH, or are in use.

Reply:
Yeah but `context["config"].banned_ports_parsed` isn't being used anymore now except checking if everything is full in L77. 

Maybe just take L83's ` banned_ports_list = []` and start it off as a deep copy of `context["config"].banned_ports_parsed`.